packages/typescript/src/core/entities/span.ts

```diff
@@ -0,0 +1,577 @@
+  }
+
+  /**
+   * Get the parent span ID
+   */
+  get parentId(): string | null {
+    return this._span.parentSpanContext?.spanId ?? null;
+  }
+
+  get name(): string {
+    return this._span.name;
+  }
+
+  get startTime(): HrTime {
+    return this._span.startTime;
+  }
+
+  get endTime(): HrTime | null {
+    return this._span.endTime;
+  }
+
+  get status(): SpanStatus {
+    return SpanStatus.fromOtelStatus(this._span.status);
+  }
+
+  get inputs(): any {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return this.getAttribute(SpanAttributeKey.INPUTS);
+  }
+
+  get outputs(): any {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return this.getAttribute(SpanAttributeKey.OUTPUTS);
+  }
+
+  get attributes(): Record<string, any> {
+    return this._attributesRegistry.getAll();
+  }
+
+  getAttribute(key: string): any {
+    return this._attributesRegistry.get(key);
+  }
+
+  get events(): SpanEvent[] {
+    return this._span.events.map((event) => {
+      const [seconds, nanoseconds] = event.time;
+      return new SpanEvent({
+        name: event.name,
+        attributes: event.attributes as Record<string, any>,
+        timestamp: seconds * 1_000_000_000 + nanoseconds
+      });
+    });
+  }
+
+  /**
+   * Convert this span to JSON format (OpenTelemetry format)
+   * @returns JSON object representation of the span
+   */
+  toJson(): any {
+    return {
+      trace_id: encodeTraceIdToBase64(this.traceId),
+      span_id: encodeSpanIdToBase64(this.spanId),
+      parent_span_id: this.parentId ? encodeSpanIdToBase64(this.parentId) : undefined,
+      name: this.name,
+      start_time_unix_nano: convertHrTimeToNanoSeconds(this.startTime),
+      end_time_unix_nano: this.endTime ? convertHrTimeToNanoSeconds(this.endTime) : null,
+      status: {
+        code: this.status?.statusCode || 'UNSET'
+      },
+      attributes: this.attributes || {},
+      events: this.events.map((event) => ({
+        name: event.name,
+        time_unix_nano: event.timestamp,
+        attributes: event.attributes || {}
+      }))
+    };
+  }
+
+  /**
+   * Create a Span from JSON data (following Python implementation)
+   * Converts the JSON data back into OpenTelemetry-compatible span
+   */
+  static fromJson(json: Record<string, any>): ISpan {
+    // Convert the JSON data back to an OpenTelemetry-like span structure
+    // This is simplified compared to Python but follows the same pattern
+
+    /* eslint-disable @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-argument */
+    const otelSpanData = {
+      name: json.name as string,
+      startTime: convertNanoSecondsToHrTime(json.start_time_unix_nano as bigint),
+      endTime: convertNanoSecondsToHrTime(json.end_time_unix_nano as bigint),
+      status: {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        code: convertStatusCodeToOTel(json.status?.code) || OTelSpanStatusCode.UNSET
```

Comment:
nit: this function seems to never return null, so the `||` should be unnecessary
```suggestion
        code: convertStatusCodeToOTel(json.status?.code)
```

Reply:
good catch!